include/boost/unordered/detail/foa/rw_spinlock.hpp

```diff
@@ -0,0 +1,188 @@
+#ifndef BOOST_UNORDERED_DETAIL_FOA_RW_SPINLOCK_HPP_INCLUDED
+#define BOOST_UNORDERED_DETAIL_FOA_RW_SPINLOCK_HPP_INCLUDED
+
+// Copyright 2023 Peter Dimov
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#include <boost/smart_ptr/detail/sp_thread_pause.hpp>
+#include <boost/smart_ptr/detail/sp_thread_sleep.hpp>
```

Comment:
what a weird place to have those functions - I can see why no one wanted to put them in `Boost.Thread`, and `Boss.Atomic` isn't right either - but still it's... weird... to be in `SmartPtr`.

Reply:
They are part of the spinlock implementation in SmartPtr (which is itself an implementation detail.)